src/core/dep-graph.ts

```diff
@@ -50,6 +51,8 @@ class DepGraphImpl implements types.DepGraphInternal {
     this._rootPkgId = (graph.node(rootNodeId) as Node).pkgId;
 
     this._pkgList = _.values(pkgs);
+    this._depPkgsList = this._pkgList
+      .filter((pkg) => pkg !== this.rootPkg);
```

Comment:
I don't really know about the internals of a graph, but could you do a `.findIndex` and remove the first occurance of the root package? That would solve the cyclic issue, assuming it appears multiple times?

Reply:
not sure I understand. It won't appear multiple times in this list even if it's cyclic

Reply:
Cool, that makes my suggestion redundant then :)